### PR TITLE
collect and report own user activity to backend

### PR DESCRIPTION
**Pull request by jjelschen**
* ~this contains a copy of `HttpProviderService`, which can be removed in favor of the implementation in [commons](https://github.com/test-editor/web-testeditor-commons).~ See [corresponding pull request](https://github.com/test-editor/web-testeditor-commons/pull/30).
   * … and now this one: https://github.com/test-editor/web-testeditor-commons/pull/31 :disappointed: 
   * removed `HttpProvider` as of 053a46a, but requires the above-mentioned changes in commons to be applied
* ~~TODO: this PR does not yet implement the handling of server responses and publishing them via message bus.~~
    * … Now it does.

### CODE REVIEW

File: src/app/modules/user-activity/user-activity.service.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import { MessagingService } from '@testeditor/messaging-service';
+import 'rxjs/add/observable/timer';
+import 'rxjs/add/operator/takeUntil';
+import { Observable } from 'rxjs/Observable';
+import { Subject } from 'rxjs/Subject';
+import { Subscription } from 'rxjs/Subscription';
+import { ElementActivity, USER_ACTIVITY_UPDATED } from '../event-types-out';
+import { HttpProviderService } from '@testeditor/testeditor-commons';
+
+export interface UserActivityEvent { name: string; elementKey: string; activityType: string; active: boolean; }
```
Comment (gunther-bachmann) [inline src/app/modules/user-activity/user-activity.service.ts]: elementKey is the key into the payload that has element key as value, right? I can't find a better name but you should probably add some documentation to UserActivityEvent to describe the fields and their roles.

File: src/app/modules/event-types-out.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+export enum UserActivityType { RUNNING_AS_TEST, OPENED_IN_EDITOR, ACTIVE_IN_EDITOR, DIRTY_IN_EDITOR, TYPED_INTO_EDITOR,
```
[the remark this answers is not in the document]
Comment (jjelschen) [reply src/app/modules/event-types-out.ts]: That's right, removed it.

File: src/app/modules/user-activity/user-activity.service.ts (hunk 1 above)
Comment (jjelschen) [reply src/app/modules/user-activity/user-activity.service.ts]: Good point, done.

**Comment by jjelschen** [comment]
**Note**: I changed the package name (used for publishing to npm) to be more in line with our other web projects (whose Git project name usually start with a "web-" prefix, but their package names do not include that prefix).
This shouldn't be a problem, since this project has never been successfully packaged and published before…

**Pull request merged**